_posts/2020-10-27-rails-upgrade-shims.md

```diff
@@ -0,0 +1,101 @@
+
+In this article I will list most of the shims that happened to in the past versions of Rails.
+
+<!--more-->
+
+## Rails 5.0
+
+### [`record_tag_helper`](https://github.com/rails/record_tag_helper) ([See changes](https://github.com/rails/rails/pull/18411))
+
+`content_tag_for` and `div_for` were removed in favor of just using `content_tag`. To continue using the older methods, add the `record_tag_helper` gem to your Gemfile.
+
+### [`rails-controller-testing`](https://github.com/rails/rails-controller-testing) ([See changes](https://github.com/rails/rails/pull/20138))
+
+`assigns` and `assert_template` were extracted to the `rails-controller-testing` gem.
+
+### [`activemodel-serializers-xml`](https://github.com/rails/activemodel-serializers-xml) ([See changes](https://github.com/rails/rails/pull/21161))
+
+`ActiveModel::Serializers::Xml` was extracted from Rails to the `activemodel-serializers-xml` gem. To continue using XML serialization, add gem `activemodel-serializers-xml` to your Gemfile.
+
+## Rails 4.2
+
+### [`responders`](https://github.com/heartcombo/responders) ([See changes](https://github.com/rails/rails/pull/16526))
+
+`respond_with` and `respond_to` methods were extracted to the `responders` gem.
+
+### [`rails-deprecated_sanitizer`](https://github.com/kaspth/rails-deprecated_sanitizer) ([See changes](http://blog.plataformatec.com.br/2014/07/the-new-html-sanitizer-in-rails-4-2/))
+
+The HTML Sanitizer logic was reimplemented, but to keep using the old behavior you can use the `rails-deprecated_sanitizer` gem.
+
+`rails-deprecated_sanitizer` is supported only for Rails 4.2.
+
+### [`rails-dom-testing`](https://github.com/rails/rails-dom-testing)
+
+The `TagAssertions` module (containing methods such as `assert_tag`), was deprecated in favor of the `assert_select` methods from the `SelectorAssertions` module, which was extracted into the `rails-dom-testing` gem.
+
+## Rails 4.1
+
+### [`activesupport-json_encoder`](https://github.com/rails/activesupport-json_encoder)
+
+Removed support for the `encode_json` hook used for encoding custom objects into JSON. This feature was extracted into the `activesupport-json_encoder` gem.
+
+Deprecated `ActiveSupport.encode_big_decimal_as_string` and `ActiveSupport::JSON::Encoding::CircularReferenceError`. These features were extracted into the `activesupport-json_encoder` gem.
+
+Changes:
+
+- [https://github.com/rails/rails/pull/12183](https://github.com/rails/rails/pull/12183)
+- [https://github.com/rails/rails/pull/12785](https://github.com/rails/rails/pull/12785)
+- [https://github.com/rails/rails/pull/13060](https://github.com/rails/rails/pull/13060)
+
+## Rails 4.0
+
+### [`protected_attributes`](https://github.com/rails/protected_attributes) ([See changes](https://github.com/rails/rails/pull/7251))
```

Comment:
maybe here you could mention `rails_upgrader` and our complete guide to migrate to strong params? https://www.fastruby.io/blog/rails/upgrades/strong-parameters-migration-guide.html

Reply:
Good idea! 👍